### PR TITLE
Move version option out of middleware

### PR DESCRIPTION
**Pull request by jonsequitur**

### CODE REVIEW

File: src/System.CommandLine/Builder/CommandLineBuilderExtensions.cs [hunk 1]
```diff
@@ -643,6 +624,7 @@ public static CommandLineBuilder UseHelp(this CommandLineBuilder builder, int? m
             builder.VersionOption = versionOption;
             builder.Command.AddOption(versionOption);
 
+#if false
```
Comment (Jozkee) [inline src/System.CommandLine/Builder/CommandLineBuilderExtensions.cs:627]:
```suggestion
```

File: src/System.CommandLine/Builder/CommandLineBuilderExtensions.cs [hunk 2]
```diff
@@ -661,6 +643,7 @@ public static CommandLineBuilder UseHelp(this CommandLineBuilder builder, int? m
                     await next(context);
                 }
             }, MiddlewareOrderInternal.VersionOption);
+#endif
```
Comment (Jozkee) [inline src/System.CommandLine/Builder/CommandLineBuilderExtensions.cs:646]:
```suggestion
```

Same for the other `#if` on line 670.

File: src/System.CommandLine/Builder/CommandLineBuilderExtensions.cs (hunk 1 above)
Comment (Jozkee) [reply src/System.CommandLine/Builder/CommandLineBuilderExtensions.cs:627]: Actually I misunderstood, this is disabling the code, right?

Do we want to keep it that way or remove the whole block?

Comment (Keboo) [reply src/System.CommandLine/Builder/CommandLineBuilderExtensions.cs:627]: I would propose removal of the whole block

**Review by adamsitnik — APPROVED** [review]
LGTM! The allocations have dropped nicely!

|        Method |      Args |   before  |     after |
|-------------- |---------- |----------:|----------:|
|  DefaultsSync | String[0] |  10,584 B |  10,264 B |
| DefaultsAsync | String[0] |  10,584 B |  10,264 B |
|   MinimalSync | String[0] |   3,256 B |   3,264 B |
|  MinimalAsync | String[0] |   3,256 B |   3,264 B |
|  DefaultsSync | String[4] |  12,080 B |  11,760 B |
| DefaultsAsync | String[4] |  12,080 B |  11,760 B |
|   MinimalSync | String[4] |   4,752 B |   4,760 B |
|  MinimalAsync | String[4] |   4,752 B |   4,760 B |

**Pull request merged**